go/common/headers.go

```diff
@@ -20,29 +20,29 @@ var hasherPool = sync.Pool{
 // Header is a public / plaintext struct that holds common properties of rollups batches.
 // Making changes to this struct will require GRPC + GRPC Converters regen
 type Header struct {
+	// The fields present in Geth's `types/Header` struct.
 	ParentHash  L2RootHash
-	Agg         common.Address
-	Nonce       types.BlockNonce // Nonce ensure compatibility with ethereum
-	L1Proof     L1RootHash       // the L1 block used by the enclave to generate the current rollup
-	Root        StateRoot
-	TxHash      common.Hash // todo - include the synthetic deposits
-	Number      *big.Int    // the rollup height
-	Bloom       types.Bloom
-	ReceiptHash common.Hash
-	Extra       []byte
-	R, S        *big.Int // signature values
-	Withdrawals []Withdrawal
+	UncleHash   common.Hash    `json:"sha3Uncles"`
```

Comment:
We have to be careful to have these JSON annotations wherever the field name diverges from what's in the JSON object returned by the Ethereum API. Otherwise, fields will end up nulled out when deserialised.